uportal-war/src/main/java/org/jasig/portal/portlets/portletadmin/PortletAdministrationHelper.java

```
@@ -158,87 +153,48 @@
-                IAuthorizationPrincipal[] prins =
-                        pm.getAuthorizedPrincipals(IPermission.PORTLET_SUBSCRIBER_ACTIVITY,
-                        PermissionHelper.permissionTargetIdForPortletDefinition(def));
-                for (int mp = 0; mp < prins.length; mp++) {
-                	JsonEntityBean bean;
-
-                	// first assume this is a group
-                	IEntityGroup group = GroupService.findGroup(prins[mp].getKey());
-                	if (group != null) {
-                    	bean = new JsonEntityBean(group, EntityEnum.GROUP);
-                	}
-
-                	// if a matching group can't be found, try to find a matching
-                	// non-group entity
-                	else {
-                    	IGroupMember member = AuthorizationService.instance().getGroupMember(prins[mp]);
-                    	bean = new JsonEntityBean(member, EntityEnum.PERSON);
-                    	String name = groupListHelper.lookupEntityName(bean);
-                    	bean.setName(name);
-                	}
-
-                    form.addGroup(bean);
-                }
-			} catch (GroupsException e) {
-				e.printStackTrace();
-			}
-		}
+            // create a JsonEntityBean for each current category and add it
+            // to our form bean's category list
+            Set<PortletCategory> categories = portletCategoryRegistry.getParentCategories(def);
+            for (PortletCategory cat : categories) {
+                form.addCategory(new JsonEntityBean(cat));
+            }
 
-        // otherwise, if this is a new portlet, pre-populate the categories
-        // and groups with some reasonable defaults
+            try {
+                getGroupPermissions(def, form, IPermission.PORTLET_SUBSCRIBER_ACTIVITY);
+                getGroupPermissions(def, form, IPermission.PORTLET_BROWSE_ACTIVITY);
+            } catch (GroupsException e) {
+                e.printStackTrace();
```

This catch block also eats the exception, do we want that behavior or a rethrow?